src/conditions/condition-expr.ts

```diff
@@ -28,13 +32,28 @@ export class ConditionExpression {
   ) {}
 
   public toObj(): ConditionExpressionJSON {
-    const conditionData = this.condition.toObj();
+    const condition = this.condition.toObj();
     return {
       version: this.version,
-      condition: conditionData,
+      condition,
     };
   }
 
+  private static conditionFromObject(obj: ConditionProps): Condition {
+    switch (obj.conditionType) {
+      case 'rpc':
+        return new RpcCondition(obj as RpcConditionProps);
+      case 'time':
+        return new TimeCondition(obj as TimeConditionProps);
+      case 'contract':
+        return new ContractCondition(obj as ContractConditionProps);
+      case 'compound':
+        return new CompoundCondition(obj as CompoundConditionProps);
+      default:
+        throw new Error(`Invalid conditionType: ${obj.conditionType}`);
+    }
+  }
```

Comment:
RFC: Do we really need access to concrete types now that we have the `conditionType` in every `Condition`? For example, what is there to gain for our users from having an `RpcCondition` instead of a `Condition` instance?

Reply:
I'm also +1 for having this simplified

Reply:
So just to reiterate: "I, as a TS developer, don't need to know the concrete type of my `condition` because I already know it's `conditionType`". Does that make sense?

Reply:
> So just to reiterate: "I, as a TS developer, don't need to know the concrete type of my condition because I already know it's conditionType". Does that make sense?

Not sure I follow. What would be the outcome of such a plan?

Reply:
Let me rephrase that: 
1) I just deserialized a condition. I want to know what type of condition it is. I can just check (by hand or programmatically) what is the value of `conditionType`. That's all good, the problem is solved.
2) But now, why would I need to know the concrete TypeScript type of my conditions? And if I ever need to know that, I follow steps in 1) and cast my condition to the desired type, `condition as RpcCondition` (or any other condition, based on `conditionType`). So, essentially what the code above is doing.

Reply:
[ ] Move to `Condition.fromObj`